modules/library/xml/src/main/java/org/geotools/xml/styling/SLDParser.java

```diff
@@ -1253,8 +1253,9 @@ protected OtherText parseOtherText(Node root) {
     private void parseVendorOption(Map<String, String> options, Node child) {
         String key = child.getAttributes().getNamedItem("name").getNodeValue();
         String value = getFirstChildValue(child);
-
-        options.put(key, value);
+        if (value != null) {
```

Comment:
I would go and make sure it's not null, not empty, and not blank, all vendor options are supposed to have some content.
Commons-lang3 is in the classpath here, using ``SystemUtils.isNotBlank`` should do the trick.